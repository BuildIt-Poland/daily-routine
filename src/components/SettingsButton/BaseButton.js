import styled from 'styled-components';

import { colorWhite } from '../../styles/designTokens';
import RadioButtonLabel from './BaseLabel';

const BaseButton = styled.input`
  display: none;
  &:checked + ${RadioButtonLabel} {
    background-color: ${({ theme }) => theme.secondaryColor};
    color: ${colorWhite};
  }
`;

export default BaseButton;
