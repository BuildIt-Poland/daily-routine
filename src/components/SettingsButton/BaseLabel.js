import styled from 'styled-components';

import { borderWidthThick, colorWhite } from '../../styles/designTokens';

const BaseLabel = styled.label`
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  height: 4rem;
  width: 100%;
  background-color: ${colorWhite};
  color: ${({ theme }) => theme.secondaryColor};
  border: ${borderWidthThick} solid ${({ theme }) => theme.secondaryColor};
`;

export default BaseLabel;
