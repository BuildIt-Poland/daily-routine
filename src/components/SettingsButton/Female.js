import styled from 'styled-components';

import { colorWhite } from '../../styles/designTokens';

const Female = styled.span`
  justify-content: center;
  align-items: center;
  display: flex;
  flex: 1;
  background-color: ${({ checked, theme }) => (checked ? 'initial' : theme.secondaryColor)};
  color: ${({ checked, theme }) => (checked ? theme.secondaryColor : colorWhite)};
`;

export default Female;
