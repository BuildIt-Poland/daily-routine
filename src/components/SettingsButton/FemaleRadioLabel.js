import styled from 'styled-components';

import { borderRadius } from '../../styles/designTokens';
import RadioLabel from './RadioLabel';

const FemaleRadioLabel = styled(RadioLabel)`
  border-radius: 0 ${borderRadius} ${borderRadius} 0;
`;

export default FemaleRadioLabel;
