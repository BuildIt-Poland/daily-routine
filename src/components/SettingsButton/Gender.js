import styled from 'styled-components';

import { borderWidthThick, borderRadius } from '../../styles/designTokens';

const Gender = styled.span`
  display: flex;
  height: 4rem;
  width: 100%;
  cursor: pointer;
  border: ${borderWidthThick} solid ${({ theme }) => theme.secondaryColor};
  border-radius: ${borderRadius};
`;

export default Gender;
