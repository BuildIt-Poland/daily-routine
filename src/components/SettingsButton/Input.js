import styled from 'styled-components';

const Input = styled.input`
  visibility: hidden;
  position: absolute;
`;
export default Input;
