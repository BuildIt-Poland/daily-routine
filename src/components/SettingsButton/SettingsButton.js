import React from 'react';
import PropTypes from 'prop-types';

import { MALE, FEMALE } from '../../constants/genders';
import RadioButton from './RadioButton';
import RadioButtonGroup from './RadioButtonGroup';
import Text from './Text';
import Form from './Form';

function SettingsButton({ gender, onChange }) {
  return (
    <Form>
      <Text>Gender</Text>
      <RadioButtonGroup>
        <RadioButton value={MALE} checked={gender === MALE} onChange={onChange}>
          Male
        </RadioButton>
        <RadioButton value={FEMALE} checked={gender === FEMALE} onChange={onChange}>
          Female
        </RadioButton>
      </RadioButtonGroup>
    </Form>
  );
}

SettingsButton.prototype = {
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired
};

export default SettingsButton;
