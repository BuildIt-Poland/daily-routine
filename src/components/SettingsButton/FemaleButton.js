import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { FEMALE } from '../../constants/genders';
import { borderRadius } from '../../styles/designTokens';
import BaseLabel from './BaseLabel';
import BaseButton from './BaseButton';

const Label = styled(BaseLabel)`
  border-radius: 0 ${borderRadius} ${borderRadius} 0;
`;

function FemaleButton({ checked, onChange }) {
  return (
    <>
      <BaseButton id="female-button" type="radio" value={FEMALE} checked={checked} onChange={onChange} />
      <Label htmlFor="female-button">Female</Label>
    </>
  );
}

FemaleButton.prototype = {
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired
};

export default FemaleButton;
