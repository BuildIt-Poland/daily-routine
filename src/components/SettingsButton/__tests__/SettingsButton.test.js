import React from 'react';
import { create } from 'react-test-renderer';

import SettingsButton from '../SettingsButton';

describe('COMPONENT - SettingsButton', () => {
  it('renders correctly', () => {
    const component = create(<SettingsButton checked="false" onChange={jest.fn()} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
