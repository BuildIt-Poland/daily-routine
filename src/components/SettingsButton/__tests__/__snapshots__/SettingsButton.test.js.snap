// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SettingsButton renders correctly 1`] = `
.c4 {
  visibility: hidden;
  position: absolute;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  height: 4rem;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid;
}

.c3:first-child {
  border-radius: 0.6rem 0 0 0.6rem;
}

.c3:last-child {
  border-radius: 0 0.6rem 0.6rem 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-weight: 700;
  margin: 0 0 1rem;
}

.c0 {
  margin: 2rem;
}

<form
  className="c0"
>
  <p
    className="c1"
  >
    Gender
  </p>
  <div
    className="c2"
  >
    <label
      checked={false}
      className="c3"
    >
      <input
        checked={false}
        className="c4"
        name="radio-button"
        onChange={[MockFunction]}
        type="radio"
        value="male"
      />
      Male
    </label>
    <label
      checked={false}
      className="c3"
    >
      <input
        checked={false}
        className="c4"
        name="radio-button"
        onChange={[MockFunction]}
        type="radio"
        value="female"
      />
      Female
    </label>
  </div>
</form>
`;
